File: test/9b45633a1769c5/rust/src/main.rs
fn collatz(mut i: usize) -> usize {
    let mut cnt = 0;
    while i != 1 {
        cnt += 1;
        if i % 2 == 0 {
            i /= 2;
        } else {
            i *= 3;
            i += 1;
        }
    }
    cnt
}
fn main() {
    for i in 0..1000000000000000000 {
        println!("{}", collatz(i));
    }
}

File: test/9b45633a1769c5/java/Main.java
class Main {
    public static long collatz(long j) {
        long i = j;
        long cnt = 0;
        while (i != 1) {
            cnt++;
            if (i % 2 == 0) {
                i /= 2;
            } else {
                i *= 3;
                i += 1;
            }
        }
        return cnt;
    }

    public static void main(String[] args) {
        for (int i = 0; i < 1000000000000000000L; i++) {
            System.out.println(collatz(i));
        }
    }
}
